.github/workflows/docker.yml

```diff
@@ -0,0 +1,90 @@
+name: Build and Push Docker Images
+
+on:
+  push:
+    branches: [ "main" ]
+    paths:
+      - 'backend/**'
+      - 'frontend/**'
+      - 'charts/kam/**'
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: '2'
+
+      - name: Install yq
+        run: |
+          sudo wget https://github.com/mikefarah/yq/releases/download/v4.35.1/yq_linux_amd64 -O /usr/bin/yq
+          sudo chmod +x /usr/bin/yq
+
+      - name: Check if appVersion changed
+        id: check_version
+        run: |
+          CHANGED_FILES=$(git diff --name-only HEAD~1 HEAD)
+
+          CURRENT_VERSION=$(yq e '.appVersion' charts/kam/Chart.yaml)
+          PREVIOUS_VERSION=$(git show HEAD~1:charts/kam/Chart.yaml | yq e '.appVersion' -)
+          echo "Current appVersion: $CURRENT_VERSION"
+          echo "Previous appVersion: $PREVIOUS_VERSION"
+          
+          if [[ ! "$CURRENT_VERSION" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "Error: appVersion must be in the format v<major>.<minor>.<patch>, e.g. v1.0.0"
+            exit 1
+          fi
+
+          if [ "$CURRENT_VERSION" == "$PREVIOUS_VERSION" ]; then
+            VERSION_CHANGED=false
+          else
+            VERSION_CHANGED=true
+          fi
+          
+          echo "::set-output name=version_changed::$VERSION_CHANGED"
+          echo "::set-output name=app_version::$CURRENT_VERSION"
+
+          if [ "$VERSION_CHANGED" = "false" ]; then
+            echo "Error: appVersion was not updated."
+            exit 1
+          else
+            echo "Validation passed."
+          fi
```

Comment:
This does not spark joy. What is the benefit of forcing us to updating app version each time we push to main? We are actively developing, I see this as slowing us down. If you want to insist on updating app version each time we push to main make the app version changing automated, by changing the small version

Reply:
Is this really that much of a problem? This only aplies when merging with main, not on every commit in pull request. So you only need to do this once. And the benefit is, we have clear docker image history and can easily run any version with helm. Plus it would be necessary if we had helm repo. And I think @samuelus wants one.

Reply:
Decision postponed; waiting for input from others.

Reply:
I don't know where exactly to address these comments, so I'll add to the already existing discussion.
1. Have you tested this pipeline that all logic works as you intend?
2. From what I understand, a change in the backend will generate a new image on the frontend and vice versa. Same for the chart update. If this is the case, then it should be separated, because it will waste a lot of pipeline time. And with such an assumption, I think some kind of auto incrementer would be needed. On the other hand, I would not exclude the use of a manual version, because in case of a large number of changes it would make sense to raise this version.

Reply:
2. You are correct, any changes in backend, frontend or charts will generate both images. This is intended. Because of helm and will not be changed.

Reply:
About docker pipeline, I propose this compromise: I will restrict pipeline only to start when charts has been changed, not backend or frontend.

Reply:
Why does helm force us to create a new frontend image when we change the default value in helm chart from “dupa” to “hehe-dupa”? This seems very stupid, and if in fact this is unavoidable, then I agree with Marek that we should not block upload to main, but only run this pipeline at the moment if someone specifically changes the versions of the application in order to upload new images. Updating everything every time we commits to main will be too long. 
Another issue. What about testing? Have you tested this, or “it should work and we will test it on the main”? XD

Reply:
> Have you tested this, or “it should work and we will test it on the main”? XD

Guess.

> Why does helm force us to create a new frontend image when we change the default value in helm chart from “dupa” to “hehe-dupa”? This seems very stupid, and if in fact this is unavoidable, then I agree with Marek that we should not block upload to main, but only run this pipeline at the moment if someone specifically changes the versions of the application in order to upload new images. 

I guess I can change it to only work when changing appVersion, but I'm not a 🪭. Not at all.

> Updating everything every time we commits to main will be too long.

Really? How often do we push to main?

Reply:
Only when updating chart and appVersion in updated